kyuubi-events/src/main/scala/org/apache/kyuubi/events/handler/ElasticSearchLoggingEventHandler.scala

```diff
@@ -0,0 +1,137 @@
+import org.apache.kyuubi.events.KyuubiEvent
+import org.apache.kyuubi.events.handler.ElasticSearchLoggingEventHandler.getElasticClient
+
+/**
+ * This event logger logs events to ElasticSearch.
+ */
+class ElasticSearchLoggingEventHandler(
+    indexId: String,
+    serverUrl: String,
+    username: Option[String] = None,
+    password: Option[String] = None,
+    kyuubiConf: KyuubiConf) extends EventHandler[KyuubiEvent] with Logging {
+
+  private val isAutoCreateIndex: Boolean =
+    kyuubiConf.get(SERVER_EVENT_ELASTICSEARCH_INDEX_AUTOCREATE_ENABLED)
+
+  private val esClient: ElasticClient = getElasticClient(serverUrl, username, password)
+
+  private def checkIndexExisted: String => Boolean = (indexId: String) => {
+    checkArgument(
+      StringUtils.isNotBlank(indexId),
+      "index name must be configured for ElasticSearchEventHandler, please ensure %s is set",
+      SERVER_EVENT_ELASTICSEARCH_INDEX.key)
+    // check
+    esClient.execute {
+      indexExists(indexId)
+    }.await.result.isExists
+  }
+
+  private def checkAndEnsureIndex(indexId: String): Unit = {
+    if (!checkIndexExisted(indexId) && isAutoCreateIndex) {
+      esClient.execute {
+        createIndex(indexId)
+          .mapping(MappingDefinition(dynamic = Some(DynamicMapping.Dynamic)))
+      }.await
+      if (!checkIndexExisted(indexId)) {
+        throwExceptionIndexNotfound(indexId)
+      }
+    } else {
+      throwExceptionIndexNotfound(indexId)
+    }
+  }
+
+  checkAndEnsureIndex(indexId)
+
+  private val objectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
+
+  override def apply(event: KyuubiEvent): Unit = {
+    val fields = {
+      objectMapper.readValue(event.toJson, classOf[Map[String, Any]]).map {
+        case (key: String, value: Any) =>
+          val isNumber = (obj: Any) => {
+            obj.isInstanceOf[Byte] || obj.isInstanceOf[Short] ||
+            obj.isInstanceOf[Int] || obj.isInstanceOf[Long] ||
+            obj.isInstanceOf[Float] || obj.isInstanceOf[Double]
+          }
+          if (value.isInstanceOf[String] || isNumber(value)) {
+            (key, value)
+          } else {
+            (key, objectMapper.writeValueAsString(value))
+          }
+      }
+    }
+    esClient.execute {
+      indexInto(indexId).fields(fields).refresh(RefreshPolicy.Immediate)
+    }.await match {
+      case f: RequestFailure =>
+        error(s"Failed to send event in ElasticSearchLoggingEventHandler, ${f.error}")
+      case _ =>
+    }
+  }
+
+  override def close(): Unit = {
+    esClient.close()
+  }
+
+  private def throwExceptionIndexNotfound(indexId: String) =
+    throw new RuntimeException(s"the index '$indexId' is not found on ElasticSearch")
+}
+
+object ElasticSearchLoggingEventHandler {
+  def getElasticClient(
+      serverUrl: String,
+      user: Option[String],
+      password: Option[String]): ElasticClient = {
+    val props = ElasticProperties(serverUrl)
+    val credentialsProvider = {
```

Comment:
Specify credentialsProvider only if username and password are defined.

Reply:
Fixed.